src/core/data/scalar.h

```diff
@@ -83,12 +82,11 @@ class Scalar {
 
  public:
   /**
-   * @brief Indicates if the `Scalar` object represents a tuple
+   * @brief Returns the data type of the scalar
    *
-   * @return true The `Scalar` is a tuple
-   * @return false The `Scalar` is a scalar
+   * @return Data type
    */
-  bool is_tuple() const { return tuple_; }
+  const Type* type() const { return type_.get(); }
```

Comment:
Could probably return a `const Type&` here, like `Store.type()` does, to avoid exposing the pointer.